app/controllers/admin/export/document_controller.rb

```diff
@@ -24,6 +27,24 @@ def edition_associations(edition)
     associations.each do |association|
       output[:associations][association] = edition.public_send(association)
     end
+    output[:whitehall_admin_links] = resolve_whitehall_admin_links(edition.body)
```

Comment:
Do you think we should start moving this code out into a presenter soon?

Also, I'm hoping this is the case but is there risk these could be in other fields than edition.body? I feel like they might be something that can be part of a withdrawn notice and maybe an unpublishing - hopefully no others 😬

Reply:
I've refactored the controller to use a presenter, which seems like a very sensible change.

I guess we'll need to audit all the places these links are included.  I'll have to think about a way of doing that.

Reply:
As far as I can see, it is only the edition body, attachments and the alternative format contact email that are processed as Govspeak (and therefore Whitehall URLs replaced).

Reply:
Withdrawn notices are done by Govspeak as well atleast:

```
app/workers/publishing_api_withdrawal_worker.rb
14:      explanation: Whitehall::GovspeakRenderer.new.govspeak_to_html(explanation),

app/workers/publishing_api_gone_worker.rb
4:      rendered_explanation = Whitehall::GovspeakRenderer
```

Reply:
I can see withdrawals and gones are too:

```
app/workers/publishing_api_withdrawal_worker.rb
14:      explanation: Whitehall::GovspeakRenderer.new.govspeak_to_html(explanation),

app/workers/publishing_api_gone_worker.rb
4:      rendered_explanation = Whitehall::GovspeakRenderer
```

there's also HTML attachments, which I guess we're probably pulling out with this data.

Reply:
Turns out we're not exporting the body of HTML attachments.  I'll write up a card for that (the Whitehall links can be dealt with at the same time).

I've addressed the withdrawal explanation in this PR.